x-pack/plugins/snapshot_restore/public/app/components/repository_verification_badge.tsx

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { RepositoryVerification } from '../../../common/types';
+import { useAppDependencies } from '../index';
+
+import { EuiHealth } from '@elastic/eui';
+
+interface Props {
+  verificationResults: RepositoryVerification | null;
+}
+
+export const RepositoryVerificationBadge: React.FunctionComponent<Props> = ({
+  verificationResults,
+}) => {
+  const {
+    core: {
+      i18n: { FormattedMessage },
+    },
+  } = useAppDependencies();
+
+  if (!verificationResults) {
+    return (
+      <EuiHealth color="subdued">
+        <FormattedMessage
+          id="xpack.snapshotRestore.repositoryVerification.verificationUnknownValue"
+          defaultMessage="Unknown"
+        />
+      </EuiHealth>
+    );
+  }
+
+  if (verificationResults.valid) {
+    return (
+      <EuiHealth color="success">
+        <FormattedMessage
+          id="xpack.snapshotRestore.repositoryVerification.verificationSuccessfulValue"
+          defaultMessage="Success"
```

Comment:
to me, "Not verified" means "we didn't check this repository" rather than that there was an error with connecting to the repository. although ES uses the term "verification", I think "repository status" or "connection status" is more accurate. let's defer this until copywriting review, but maybe we can simply say "Status" instead, maybe with the values "Connected" or "Error"